pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -97,11 +97,11 @@ def get_reduced_metrics(self):
         return self._internals_reduced
 
     def add_dataloader_idx(self):
-        return len(self._internals) > 1
+        return len(self._internals_reduced) > 1 if self.has_reduced else len(self._internals) > 1
```

Comment:
use already defined prop
```suggestion
        return self.num_dataloaders > 1
```
pls add typing :]